### PR TITLE
feat(dlang): create module

### PR DESCRIPTION
**Pull request by Milo123459**
#### Description

Adds a module for dlang
#### Motivation and Context

Closes #

#### Screenshots (if appropriate):
img
#### How Has This Been Tested?

- [ ] I have tested using **MacOS**
- [ ] I have tested using **Linux**
- [x] I have tested using **Windows**

#### Checklist:

- [x] I have updated the documentation accordingly.
- [x] I have updated the tests accordingly.

### CODE REVIEW

File: src/modules/dlang.rs [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+        .split('\n')
+        .next()?
+        .split_whitespace()
+        .nth(3)?;
+
+    Some(version.to_owned())
+}
+
+#[cfg(test)]
+mod tests {
```
Comment (vladimyr) [inline src/modules/dlang.rs:61]: Please add tests for dub manifests: `dub.json` and `dub.sdl`
https://dub.pm/package-format-json.html
https://dub.pm/package-format-sdl.html

**Review by vladimyr — CHANGES_REQUESTED** [review]

File: docs/config/README.md [hunk 2]
```diff
@@ -686,6 +686,42 @@ By default the module will be shown if any of the following conditions are met:
 format = "via [🦕 $version](green bold) "
 ```
 
+## DLang
+
+The `dlang` module shows you your currently installed version of D.
+By default the module will be shown if any of the following conditions are met:
+- The current directory contains a file ending with `.d` or `.di`.
+- The current directory contains a `dub.json` or `dub.sdl`.
```
Comment (vladimyr) [inline docs/config/README.md:694]:
```suggestion
- The current directory contains a file with the `.d` or `.di` extension.
- The current directory contains a `dub.json` or `dub.sdl` file.
```

File: src/modules/dlang.rs [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+}
+
+fn parse_dlang_version(dlang_version: &str) -> Option<String> {
+    let version = dlang_version
+        // split into lines
+        .lines()
+        // first line
+        .next()?
+        // split by whitespace
+        .split_whitespace()
```
Comment (vladimyr) [inline src/modules/dlang.rs:58]: I meant something along these lines https://github.com/starship/starship/blob/fe030c2064e20f1359ab4844d3b3472a1ed1f6b6/src/modules/ruby.rs#L59-L67

File: src/modules/dlang.rs [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+    use super::parse_dlang_version;
+    use crate::test::ModuleRenderer;
+    use ansi_term::Color;
+    use std::fs::File;
+    use std::io;
+
+    #[test]
+    fn test_parse_dlang_version() {
+        const OUTPUT: &str = "DMD32 D Compiler v2.096.0-dirty
+        Copyright (C) 1999-2021 by The D Language Foundation, All Rights Reserved written by Walter Bright\n";
```
Comment (vladimyr) [inline src/modules/dlang.rs:76]: Is this supposed to be indented?

File: src/modules/dlang.rs (hunk 3 above)
Comment (Milo123459) [reply src/modules/dlang.rs:58]: oh

File: src/modules/dlang.rs (hunk 4 above)
Comment (Milo123459) [reply src/modules/dlang.rs:76]: I'm confused, that's what dmd --version returns.

Comment (vladimyr) [reply src/modules/dlang.rs:76]: I'm asking about the space between dirty and copyright.
```diff
+        const OUTPUT: &str = "DMD32 D Compiler v2.096.0-dirty
+        Copyright (C) 1999-2021 by The D Language Foundation, All Rights Reserved written by Walter Bright\n";
```

Comment (Milo123459) [reply src/modules/dlang.rs:76]: Oh it's a new line

**Review by vladimyr — COMMENTED** [review]
So far so good but D has 3 major compilers that are widely used so we need a way to support all of them: `dmd`, `gdc`, and `ldc` 
https://wiki.dlang.org/Compilers

**Pull request closed without merging**